lib/ansible/modules/storage/purestorage/purefa_facts.py

```diff
@@ -585,12 +585,17 @@ def generate_host_dict(array):
     hosts = array.list_hosts()
     for host in range(0, len(hosts)):
         hostname = hosts[host]['name']
+        tports = []
+        host_all_info = array.get_host(hostname, all=True)
+        if host_all_info:
+            tports = host_all_info[0]['target_ports']
```

Comment:
Now getting a "KeyError" on "target_ports" .  should be "target_port" . ? 

line 591, in generate_host_dict\r\nKeyError: 'target_ports'\r\n"

```
{
    "host_wwn": "20000025B5000A02",
    "name": "esxi1",
    "host_iqn": null,
    "vol": "FlashStack-Datastore2",
    "hgroup": "FlashStack-Cluster",
    "target_port": [
      "CT1.FC2",
      "CT0.FC2"
    ],
    "lun": 254
```

Reply:
Now that is what you get for rushing :)